### PR TITLE
enhance uart ns16550 driver to get clock frequency from clock manager

### PR DESCRIPTION
**Pull request by teikheng**
This patch is to extend UART ns16550 driver support to get clock frequency from device tree or clock manager based on the configuration in device tree. The driver will prioritize to get the valid clock frequency defined in devicetree first, otherwise, it will get the clock frequency from clock manager.

### CODE REVIEW

File: drivers/serial/uart_ns16550.c [hunk 1]
```diff
@@ -367,7 +371,24 @@ static int uart_ns16550_configure(const struct device *dev,
...
 
-	set_baud_rate(dev, cfg->baudrate);
+	/*
+	 * set clock frequency from clock_frequency property if valid,
+	 * otherwise, get clock frequency from clock manager
+	 */
+	if (dev_cfg->sys_clk_freq != 0U) {
+		pclk = dev_cfg->sys_clk_freq;
+	} else {
+		clk_dev = device_get_binding(dev_cfg->clock_drv);
```
Comment (dcpleung) [inline drivers/serial/uart_ns16550.c:381]: We are moving away from `device_get_binding()` to `DEVICE_DT_GET()`. You can refer to commit 93f2fedc12c0ad4e2e666415a498b407fcbaa3c6 on similar changes related to clock control.

Comment (teikheng) [reply drivers/serial/uart_ns16550.c:381]: ok. I will make the change accordingly

**Comment by teikheng** [comment]
sorry I accidentally deleted the branch. I will recreate a new pull request

**Comment by teikheng** [comment]
> sorry I accidentally deleted the branch. I will recreate a new pull request

The is the new pull request link https://github.com/zephyrproject-rtos/zephyr/pull/47199

**Pull request closed without merging**